### PR TITLE
[CI] cleanup and script refinements

### PR DESCRIPTION
**Pull request by larroy**
Add group inside container
Use env in shebang
Change into_container to interactive to match docker syntax

## Description ##
(Brief description on what this PR is about)

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### CODE REVIEW

File: ci/docker/Dockerfile.build.ubuntu_base_cpu [hunk 1]
```diff
@@ -25,13 +25,13 @@ WORKDIR /work/deps
...
-ARG USER_ID=os.getuid()
-
+ARG USER_ID=0
+ARG GROUP_ID=0
 COPY install/ubuntu_adduser.sh /work/
 RUN /work/ubuntu_adduser.sh
 
-COPY install/ubuntu_runas_sudo.sh /work/
-RUN /work/ubuntu_runas_sudo.sh
+copy install/ubuntu_runas_sudo.sh /work/
```
Comment (KellenSunderland) [inline ci/docker/Dockerfile.build.ubuntu_base_cpu]: Why'd you lowercase this one?

File: ci/test_docker_cache.py [hunk 2]
```diff
@@ -251,5 +251,5 @@ def _assert_docker_build(lambda_func, expected_cache_hit_count: int, expected_ca
 
 
 if __name__ == '__main__':
-    import nose2
-    nose2.main()
+    import nose
```
Comment (lebeg) [inline ci/test_docker_cache.py]: Why is this better?

Comment (larroy) [reply ci/test_docker_cache.py]: we use nose everywhere, not nose2

File: ci/docker/Dockerfile.build.ubuntu_base_cpu (hunk 1 above)
Comment (larroy) [reply ci/docker/Dockerfile.build.ubuntu_base_cpu]: seems an editor mistake, will fix. Thanks for catching.

File: ci/build.py [hunk 3]
```diff
@@ -81,6 +81,7 @@ def build_docker(platform: str, docker_binary: str, registry: str) -> None:
     cmd = [docker_binary, "build",
            "-f", get_dockerfile(platform),
            "--build-arg", "USER_ID={}".format(os.getuid()),
+           "--build-arg", "GROUP_ID={}".format(os.getgid()),
```
[the remark this answers is not in the document]
Comment (larroy) [reply ci/build.py]: done

**Comment by larroy** [comment]
Any more comments? can we please merge this?

File: ci/docker/Dockerfile.build.ubuntu_cpu [hunk 4]
```diff
@@ -48,9 +48,13 @@ COPY install/ubuntu_docs.sh /work/
 RUN /work/ubuntu_docs.sh
 
 ARG USER_ID=0
+ARG GROUP_ID=0
 COPY install/ubuntu_adduser.sh /work/
 RUN /work/ubuntu_adduser.sh
 
+COPY install/ubuntu_runas_sudo.sh /work/
```
Comment (marcoabreu) [inline ci/docker/Dockerfile.build.ubuntu_cpu]: Please don't add this. We specifically don't want to run with sudo

Comment (larroy) [reply ci/docker/Dockerfile.build.ubuntu_cpu]: good catch

**Comment by larroy** [comment]
thanks for approving, more importantly, can we merge?

**Pull request merged**